pennylane/ops/op_math/decompositions/unitary_decompositions.py

```diff
@@ -760,3 +761,137 @@ def two_qubit_decomp_rule(U, wires, **__):
     )(U, wires, initial_phase)
     total_phase = initial_phase + additional_phase
     ops.GlobalPhase(-total_phase)
+
+
+def _compute_udv(a, b):
+    r"""Given the matrices `a` and `b`, calculates the matrices `u`, `d` and `v`
+    of Eq. 36 in [arXiv-quant-ph:0504100](https://arxiv.org/pdf/quant-ph/0504100):
+
+    .. math::
+
+        a = u d v \\
+        b = u d^{\dagger} v.
+    """
+
+    # Calculates u and d diagonalizing ab^dagger (Eq.39)
+    ab_dagger = a @ math.conj(b.T)
+    d_square, u = math.linalg.eig(ab_dagger)
+    u, _ = math.linalg.qr(u)
+
+    # complex square root of eigenvalues
+    d = math.exp(1j * math.angle(d_square) / 2)
+
+    # Calculates v using Eq.40
+    v = math.conj(math.diag(d).T) @ math.conj(u.T) @ a
+
+    return u, d, v
+
+
+def multi_qubit_decomposition(U, wires):
+    r"""Decompose a multi-qubit unitary :math:`U` in terms of elementary operations.
+
+    The n-qubit unitary :math:`U`, with :math:`n > 1`, is decomposed into four (:math:`n-1`)-qubit
+    unitaries (:class:`~pennylane.ops.QubitUnitary`) and three multiplexers (:class:`~pennylane.templates.SelectPauliRot`)
+    using the cosine-sine decomposition.
+    This implementation is based on `arXiv:quant-ph/0504100 <https://arxiv.org/pdf/quant-ph/0504100>`__.
+
+    Args:
+        U (tensor): A :math:`2^n \times 2^n` unitary matrix with :math:`n > 1`.
+        wires (Union[Wires, Sequence[int] or int]): The wires on which to apply the operation.
+
+    Returns:
+        list[Operation]: A list of operations that represent the decomposition
+        of the matrix U.
+
+    **Example**
+
+    .. code-block:: pycon
+
+        >>> matrix_target = qml.matrix(qml.QFT([0,1,2]))
+        >>> ops = multi_qubit_decomposition(matrix_target, [0,1,2])
```

Comment:
Please put the full path, i.e., `qml.ops.multi_qubit_decomposition`, otherwise there will be an error. The assumed imports should be only pennylane as qml and numpy as np.